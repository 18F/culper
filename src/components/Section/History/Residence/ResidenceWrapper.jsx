import React from 'react'
import PropTypes from 'prop-types'

import i18n from 'util/i18n'

import { Field, Show } from 'components/Form'
import { sectionHasGaps } from 'components/Section/History/helpers'

import { HISTORY, HISTORY_RESIDENCE } from 'config/formSections/history'
import * as formConfig from 'config/forms'

import ConnectedResidence from './Residence'
import ResidenceSummaryProgress from './ResidenceSummaryProgress'
import connectHistorySection from '../HistoryConnector'

const sectionConfig = {
  section: HISTORY.name,
  subsection: HISTORY_RESIDENCE.name,
}

export const ResidenceWrapper = (props) => {
  const { Residence, Birthdate, formType } = props

  const years = formType
    && formConfig[formType]
    && formConfig[formType].HISTORY_RESIDENCE_YEARS

  const formName = formType

  let residenceItems = []
  if (Residence && Residence.List && Residence.List.items) {
    residenceItems = Residence.List.items
  }

  const residenceHasGaps = sectionHasGaps(residenceItems, years)

  return (
    <div>
      <h1 className="section-header">
        {i18n.t('history.residence.title')}
      </h1>

      <Field
        title={i18n.t('history.residence.info', { years })}
        titleSize="h3"
        optional
        className="no-margin-bottom"
      >
        {i18n.m('history.residence.info2')}
        {i18n.m('history.residence.info3a')}
        {i18n.m('history.residence.info3b')}
        {i18n.m('history.residence.info3c')}
        {i18n.m('history.residence.info3d')}
      </Field>

      <span id="scrollToHistory" />

      <ResidenceSummaryProgress
        Residence={Residence}
        Birthdate={Birthdate}
        years={years}
      />

      <ConnectedResidence
        realtime
        scrollToTop="scrollToHistory"
        totalYears={years}
      />

      <Show when={residenceHasGaps}>
        <div className="not-complete">
          <hr className="section-divider" />

          <Field
            title={i18n.t('history.residence.heading.exiting')}
            titleSize="h4"
            optional
            className="no-margin-bottom"
          >
            {i18n.m('history.residence.para.exiting', { years, formName })}
          </Field>
        </div>
      </Show>
    </div>
  )
}

/* eslint react/forbid-prop-types: 0 */
ResidenceWrapper.propTypes = {
  Residence: PropTypes.object,
  Birthdate: PropTypes.any,
  formType: PropTypes.string.isRequired,
}

ResidenceWrapper.defaultProps = {
  Residence: undefined,
  Birthdate: undefined,
}

export default connectHistorySection(ResidenceWrapper, sectionConfig)
