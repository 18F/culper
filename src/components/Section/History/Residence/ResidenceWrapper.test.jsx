import React from 'react'
import { shallow } from 'enzyme'

import { ResidenceWrapper } from './ResidenceWrapper'

describe('The ResidenceWrapper component', () => {
  it('renders without errors', () => {
    const component = shallow(
      <ResidenceWrapper />
    )

    expect(component.exists()).toBe(true)
    expect(component).toMatchSnapshot()
  })
})
