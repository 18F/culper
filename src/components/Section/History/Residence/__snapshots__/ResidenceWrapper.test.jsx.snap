// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ResidenceWrapper component renders without errors 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1
      className="section-header"
    >
      Where you have lived
    </h1>
    <Field
      adjustFor=""
      className="no-margin-bottom"
      comments={false}
      commentsActive={false}
      commentsAdd="comments.add"
      commentsName="Comments"
      commentsRemove="comments.remove"
      commentsRequired={false}
      commentsValue={Object {}}
      errorPrefix=""
      errors={Array []}
      filterErrors={[Function]}
      help=""
      helpActive={false}
      optional={true}
      optionalText=""
      scrollIntoView={true}
      shrink={false}
      title="List the places where you have lived beginning with your present residence and working back  years."
      titleSize="h3"
      validate={true}
    >
      <ReactMarkdown
        containerTagName="div"
        parserOptions={Object {}}
        renderers={
          Object {
            "link": [Function],
          }
        }
        source="Residences for the entire period must be accounted for without breaks."
      />
      <ReactMarkdown
        containerTagName="div"
        parserOptions={Object {}}
        renderers={
          Object {
            "link": [Function],
          }
        }
        source=" - **Indicate the actual physical location of your residence**, not a Post Office box or a permanent residence when you were not physically located there."
      />
      <ReactMarkdown
        containerTagName="div"
        parserOptions={Object {}}
        renderers={
          Object {
            "link": [Function],
          }
        }
        source=" - **If you split your time between one or more residences during a time period**, you must list all residences."
      />
      <ReactMarkdown
        containerTagName="div"
        parserOptions={Object {}}
        renderers={
          Object {
            "link": [Function],
          }
        }
        source=" - **Do not list residences before your 18th birthday** unless to provide a minimum of 2 years residence history."
      />
      <ReactMarkdown
        containerTagName="div"
        parserOptions={Object {}}
        renderers={
          Object {
            "link": [Function],
          }
        }
        source=" - **You are not required to list temporary locations of less than 90 days** that did not serve as your permanent or mailing address."
      />
    </Field>
    <span
      id="scrollToHistory"
    />
    <ResidenceSummaryProgress
      years={10}
    />
    <Connect(ConnectedHistorySection)
      realtime={true}
      scrollToTop="scrollToHistory"
    />
    <Show
      when={true}
    >
      <div
        className="not-complete"
      >
        <hr
          className="section-divider"
        />
        <Field
          adjustFor=""
          className="no-margin-bottom"
          comments={false}
          commentsActive={false}
          commentsAdd="comments.add"
          commentsName="Comments"
          commentsRemove="comments.remove"
          commentsRequired={false}
          commentsValue={Object {}}
          errorPrefix=""
          errors={Array []}
          filterErrors={[Function]}
          help=""
          helpActive={false}
          optional={true}
          optionalText=""
          scrollIntoView={true}
          shrink={false}
          title="Before you leave this section"
          titleSize="h4"
          validate={true}
        >
          <ReactMarkdown
            containerTagName="div"
            parserOptions={Object {}}
            renderers={
              Object {
                "link": [Function],
              }
            }
            source="**The full  year period of residence history is not covered.** Your  cannot be submitted until all  years are covered with no gaps.<br><br>We will mark the gaps and highlight them for you when you come back."
          />
        </Field>
      </div>
    </Show>
  </div>,
  "nodes": Array [
    <div>
      <h1
        className="section-header"
      >
        Where you have lived
      </h1>
      <Field
        adjustFor=""
        className="no-margin-bottom"
        comments={false}
        commentsActive={false}
        commentsAdd="comments.add"
        commentsName="Comments"
        commentsRemove="comments.remove"
        commentsRequired={false}
        commentsValue={Object {}}
        errorPrefix=""
        errors={Array []}
        filterErrors={[Function]}
        help=""
        helpActive={false}
        optional={true}
        optionalText=""
        scrollIntoView={true}
        shrink={false}
        title="List the places where you have lived beginning with your present residence and working back  years."
        titleSize="h3"
        validate={true}
      >
        <ReactMarkdown
          containerTagName="div"
          parserOptions={Object {}}
          renderers={
            Object {
              "link": [Function],
            }
          }
          source="Residences for the entire period must be accounted for without breaks."
        />
        <ReactMarkdown
          containerTagName="div"
          parserOptions={Object {}}
          renderers={
            Object {
              "link": [Function],
            }
          }
          source=" - **Indicate the actual physical location of your residence**, not a Post Office box or a permanent residence when you were not physically located there."
        />
        <ReactMarkdown
          containerTagName="div"
          parserOptions={Object {}}
          renderers={
            Object {
              "link": [Function],
            }
          }
          source=" - **If you split your time between one or more residences during a time period**, you must list all residences."
        />
        <ReactMarkdown
          containerTagName="div"
          parserOptions={Object {}}
          renderers={
            Object {
              "link": [Function],
            }
          }
          source=" - **Do not list residences before your 18th birthday** unless to provide a minimum of 2 years residence history."
        />
        <ReactMarkdown
          containerTagName="div"
          parserOptions={Object {}}
          renderers={
            Object {
              "link": [Function],
            }
          }
          source=" - **You are not required to list temporary locations of less than 90 days** that did not serve as your permanent or mailing address."
        />
      </Field>
      <span
        id="scrollToHistory"
      />
      <ResidenceSummaryProgress
        years={10}
      />
      <Connect(ConnectedHistorySection)
        realtime={true}
        scrollToTop="scrollToHistory"
      />
      <Show
        when={true}
      >
        <div
          className="not-complete"
        >
          <hr
            className="section-divider"
          />
          <Field
            adjustFor=""
            className="no-margin-bottom"
            comments={false}
            commentsActive={false}
            commentsAdd="comments.add"
            commentsName="Comments"
            commentsRemove="comments.remove"
            commentsRequired={false}
            commentsValue={Object {}}
            errorPrefix=""
            errors={Array []}
            filterErrors={[Function]}
            help=""
            helpActive={false}
            optional={true}
            optionalText=""
            scrollIntoView={true}
            shrink={false}
            title="Before you leave this section"
            titleSize="h4"
            validate={true}
          >
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source="**The full  year period of residence history is not covered.** Your  cannot be submitted until all  years are covered with no gaps.<br><br>We will mark the gaps and highlight them for you when you come back."
            />
          </Field>
        </div>
      </Show>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ResidenceWrapper />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "Birthdate": undefined,
          "Residence": undefined,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h1
            className="section-header"
          >
            Where you have lived
          </h1>
          <Field
            adjustFor=""
            className="no-margin-bottom"
            comments={false}
            commentsActive={false}
            commentsAdd="comments.add"
            commentsName="Comments"
            commentsRemove="comments.remove"
            commentsRequired={false}
            commentsValue={Object {}}
            errorPrefix=""
            errors={Array []}
            filterErrors={[Function]}
            help=""
            helpActive={false}
            optional={true}
            optionalText=""
            scrollIntoView={true}
            shrink={false}
            title="List the places where you have lived beginning with your present residence and working back  years."
            titleSize="h3"
            validate={true}
          >
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source="Residences for the entire period must be accounted for without breaks."
            />
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source=" - **Indicate the actual physical location of your residence**, not a Post Office box or a permanent residence when you were not physically located there."
            />
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source=" - **If you split your time between one or more residences during a time period**, you must list all residences."
            />
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source=" - **Do not list residences before your 18th birthday** unless to provide a minimum of 2 years residence history."
            />
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source=" - **You are not required to list temporary locations of less than 90 days** that did not serve as your permanent or mailing address."
            />
          </Field>
          <span
            id="scrollToHistory"
          />
          <ResidenceSummaryProgress
            years={10}
          />
          <Connect(ConnectedHistorySection)
            realtime={true}
            scrollToTop="scrollToHistory"
          />
          <Show
            when={true}
          >
            <div
              className="not-complete"
            >
              <hr
                className="section-divider"
              />
              <Field
                adjustFor=""
                className="no-margin-bottom"
                comments={false}
                commentsActive={false}
                commentsAdd="comments.add"
                commentsName="Comments"
                commentsRemove="comments.remove"
                commentsRequired={false}
                commentsValue={Object {}}
                errorPrefix=""
                errors={Array []}
                filterErrors={[Function]}
                help=""
                helpActive={false}
                optional={true}
                optionalText=""
                scrollIntoView={true}
                shrink={false}
                title="Before you leave this section"
                titleSize="h4"
                validate={true}
              >
                <ReactMarkdown
                  containerTagName="div"
                  parserOptions={Object {}}
                  renderers={
                    Object {
                      "link": [Function],
                    }
                  }
                  source="**The full  year period of residence history is not covered.** Your  cannot be submitted until all  years are covered with no gaps.<br><br>We will mark the gaps and highlight them for you when you come back."
                />
              </Field>
            </div>
          </Show>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h1
            className="section-header"
          >
            Where you have lived
          </h1>
          <Field
            adjustFor=""
            className="no-margin-bottom"
            comments={false}
            commentsActive={false}
            commentsAdd="comments.add"
            commentsName="Comments"
            commentsRemove="comments.remove"
            commentsRequired={false}
            commentsValue={Object {}}
            errorPrefix=""
            errors={Array []}
            filterErrors={[Function]}
            help=""
            helpActive={false}
            optional={true}
            optionalText=""
            scrollIntoView={true}
            shrink={false}
            title="List the places where you have lived beginning with your present residence and working back  years."
            titleSize="h3"
            validate={true}
          >
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source="Residences for the entire period must be accounted for without breaks."
            />
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source=" - **Indicate the actual physical location of your residence**, not a Post Office box or a permanent residence when you were not physically located there."
            />
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source=" - **If you split your time between one or more residences during a time period**, you must list all residences."
            />
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source=" - **Do not list residences before your 18th birthday** unless to provide a minimum of 2 years residence history."
            />
            <ReactMarkdown
              containerTagName="div"
              parserOptions={Object {}}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              source=" - **You are not required to list temporary locations of less than 90 days** that did not serve as your permanent or mailing address."
            />
          </Field>
          <span
            id="scrollToHistory"
          />
          <ResidenceSummaryProgress
            years={10}
          />
          <Connect(ConnectedHistorySection)
            realtime={true}
            scrollToTop="scrollToHistory"
          />
          <Show
            when={true}
          >
            <div
              className="not-complete"
            >
              <hr
                className="section-divider"
              />
              <Field
                adjustFor=""
                className="no-margin-bottom"
                comments={false}
                commentsActive={false}
                commentsAdd="comments.add"
                commentsName="Comments"
                commentsRemove="comments.remove"
                commentsRequired={false}
                commentsValue={Object {}}
                errorPrefix=""
                errors={Array []}
                filterErrors={[Function]}
                help=""
                helpActive={false}
                optional={true}
                optionalText=""
                scrollIntoView={true}
                shrink={false}
                title="Before you leave this section"
                titleSize="h4"
                validate={true}
              >
                <ReactMarkdown
                  containerTagName="div"
                  parserOptions={Object {}}
                  renderers={
                    Object {
                      "link": [Function],
                    }
                  }
                  source="**The full  year period of residence history is not covered.** Your  cannot be submitted until all  years are covered with no gaps.<br><br>We will mark the gaps and highlight them for you when you come back."
                />
              </Field>
            </div>
          </Show>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ResidenceWrapper />,
}
`;
